### PR TITLE
Feature1 Master Commit

### PR DESCRIPTION
**Pull request by Tingedleaf**
hi. hope this is right pull request :)

### CODE REVIEW

**Comment by AlexBulankou** [comment]
Thanks! will take a look before EOD Pacific time

File: src/places/Yojowa.StopByStop.Places/PlacesLoader.cs [hunk 1]
```diff
@@ -1,105 +1,146 @@
...
-        {
-            List<GeoPlace> cities = new List<GeoPlace>();
-            HashSet<string> insertedCityIds = new HashSet<string>();
-            HashSet<string> unsupportedStates = new HashSet<string>(new string[] { "ak", "hi" });
 
-            using (var stream = typeof(PlacesLoader).Assembly.GetManifestResourceStream("Yojowa.StopByStop.Places.cities1000.csv"))
-            {
-                using (var streamReader = new StreamReader(stream))
+                // Re-create table
+                string createTableQuery = @"DROP TABLE IF EXISTS tng_cities_test;
```
Comment (AlexBulankou) [inline src/places/Yojowa.StopByStop.Places/PlacesLoader.cs]: Can you name the table ``cities``

File: src/places/Yojowa.StopByStop.Places/PlacesLoader.cs [hunk 2]
```diff
@@ -1,105 +1,146 @@
...
-        }
-    }
-}
+            }
+        }
+
+        /// <summary>
+        /// Creates the table.
+        /// </summary>
+        internal static void CreateTable()
```
Comment (AlexBulankou) [inline src/places/Yojowa.StopByStop.Places/PlacesLoader.cs]: Can you remove this method? This was a temporary stub, no need for it now.

File: src/places/Yojowa.StopByStop.Places/PlacesLoader.cs [hunk 3]
```diff
@@ -1,105 +1,146 @@
...
+                // Re-create table
+                string createTableQuery = @"DROP TABLE IF EXISTS tng_cities_test;
+                                                    CREATE TABLE tng_cities_test
+                                                    (
+                                                        id varchar(250) NOT NULL,
+                                                        shortname varchar(250) NOT NULL,
+                                                        name varchar(250) NOT NULL,
+                                                        lat numeric(9, 6) NOT NULL,
+                                                        lng numeric(9, 6) NOT NULL,
+                                                        population bigserial NOT NULL
```
Comment (AlexBulankou) [inline src/places/Yojowa.StopByStop.Places/PlacesLoader.cs]: why bigserial? bigserial is auto-incrementing numeral intended for ids?

File: test/unit/PlacesTest.cs [hunk 4]
```diff
@@ -1,41 +1,50 @@
...
+            PlacesLoader.CreateTable();
+        }
+
+        /// <summary>
+        /// Verifies the table is created without errors
+        /// </summary>
+        [TestMethod]
+        public void VerifyCleanAndReloadDbWithoutErrors()
+        {
+            PlacesLoader.CleanAndReloadDb();
```
Comment (AlexBulankou) [inline test/unit/PlacesTest.cs]: Two problems with this unit test:
1. It recreates production DB. Instead can you change it to be able to pass table name so it creates a separate table? 2. 2. It doesn't verify the entries that got created. Can you add this to the unit test.

File: src/places/Yojowa.StopByStop.Places/PlacesLoader.cs (hunk 1 above)
Comment (Tingedleaf) [inline src/places/Yojowa.StopByStop.Places/PlacesLoader.cs]: done.

File: src/places/Yojowa.StopByStop.Places/PlacesLoader.cs (hunk 3 above)
Comment (Tingedleaf) [inline src/places/Yojowa.StopByStop.Places/PlacesLoader.cs]: changed to bigint

File: test/unit/PlacesTest.cs (hunk 4 above)
Comment (Tingedleaf) [inline test/unit/PlacesTest.cs]: done

**Comment by Tingedleaf** [comment]
Changes done.

**Comment by AlexBulankou** [comment]
I can still see .suo file and files under .vs folder as part of the change. Can you please remove

**Comment by AlexBulankou** [comment]
Also please fix the build and unit tests - you can see they are failing with your change in the second iteration (everything passed with first iteration)

File: test/unit/packages.config [hunk 5]
```diff
@@ -2,6 +2,6 @@
 <packages>
   <package id="Microsoft.ApplicationInsights" version="2.2.0-beta1" targetFramework="net46" />
   <package id="Newtonsoft.Json" version="9.0.1" targetFramework="net46" />
-  <package id="Npgsql" version="3.0.5" targetFramework="net46" />
+  <package id="Npgsql" version="3.1.8" targetFramework="net46" />
```
Comment (AlexBulankou) [inline test/unit/packages.config]: If you don't require this update, no need to update to newer Npgsql version here

**Pull request closed without merging**